### ISSUE TITLE
New cop: `Style/ArrayDeconstruction`

### ISSUE THREAD

**Issue by fatkodima** [rubocop/rubocop#11394]
As suggested https://github.com/rubocop/rubocop/pull/11279#discussion_r1048059093

Example from rubocop:
```ruby
# bad
predecessor = tokens[index]
operator = tokens[index + 1]
successor = tokens[index + 2]

# good
predecessor, operator, successor = processed_source.tokens[index, 3]
```

Example from discourse: https://github.com/discourse/discourse/blob/7ecf4d12a96cff5044905b231cd30a3ed592e3e6/lib/tiny_japanese_segmenter.rb#L86-L97

It should check for consecutive method calls like: 
1. `Array#[]` 
2. `Array#slice`
3. `String#[]`
4.  `String#slice`
5. `String#byteslice`

Or just 1 and 3 (if that would simplify greatly the implementation) as the most popular and likely to cause the offense.
